lib/expand/join.go

```diff
@@ -158,36 +162,24 @@ func (p *Peer) Execute(req *installpb.ExecuteRequest, stream installpb.Agent_Exe
 			if err != nil {
 				return err
 			}
-		case err := <-p.execDoneC:
-			if err != nil {
+		case result := <-p.execDoneC:
+			if result.Err != nil {
 				// Phase finished with an error.
-				if installpb.IsRPCError(err) {
-					return trace.Unwrap(err)
+				if installpb.IsRPCError(result.Err) {
+					return trace.Unwrap(result.Err)
+				}
+				// Flag a failed intermediate step as codes.Aborted.
+				// From gRPC documentation:
+				// Use Aborted if the client should retry at a higher-level
+				// (e.g., restarting a read-modify-write sequence).
+				// See https://github.com/grpc/grpc-go/blob/v1.22.0/codes/codes.go#L78
+				return status.Error(codes.Aborted, trace.UserMessage(result.Err))
+			}
+			if result.CompletionEvent != nil {
```

Comment:
I updated expand side of the service to match that of installer w.r.t handling of completion status.